datastream-server/src/main/java/com/linkedin/datastream/server/Coordinator.java

```diff
@@ -710,10 +710,17 @@ private DatastreamTask getDatastreamTask(String taskName) {
     });
   }
 
-  private void uninitializeTask(DatastreamTask t) {
-    TransportProviderAdmin tpAdmin = _transportProviderAdmins.get(t.getTransportProviderName());
-    tpAdmin.unassignTransportProvider(t);
-    _cpProvider.unassignDatastreamTask(t);
+  private void uninitializeTask(List<DatastreamTask> t) {
```

Comment:
not you fault, but can you change the input variable to be "tasks" instead of "t". It's much easier to read.